### PR TITLE
652 create new labs page

### LINKED ISSUE #652: Create New Labs Page
# Create New Labs Page

## Work to Be Completed
- [x] Create new Route from <Labs/> page in App.js
- [x] Implement new _**Explore Our Labs**_ header for the Labs Page
- [x] Create Labs Section that displays labs - Can use search bar to search lab name/type, and filter options
- [x] Create _**View Your Progress**_ section that links to user's account - If user is not logged in, direct them to log in modal that is used in the header, else send them to the profile page
- [x] Create/Reuse the **_Interested in Getting Involved?_** section from the home page

Screenshot 2024-10-23 at 10 24 43 AM
Screenshot 2024-10-23 at 10 25 06 AM

### PR DESCRIPTION
**Pull request by mpb8984**
New labs page with ability to search and filter labs by category, difficulty and search phrases.

- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

File: client/src/components/body/labspage/LabsPage.js [hunk 1]
```diff
@@ -0,0 +1,386 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable no-inner-declarations */
```
Comment (heathermoses) [inline client/src/components/body/labspage/LabsPage.js]: Remove this and add prop types at the bottom

File: client/src/components/body/labspage/LabsPage.js [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+                <div className="md:lg:tw-flex tw-flex-col md:lg:tw-justify-center sm:tw-grid-cols-2 tw-flex-wrap">
+                  {Array.from(displayedLabs.entries()).map(
+                    ([category, labArray]) => (
+                      <div
+                        key={category}
+                        className="tw-flex tw-flex-col tw-mb-4"
+                      >
+                        <text className="tw-font-bold tw-font-calibri tw-text-xl tw-w-full tw-text-left">
+                          {category}
+                        </text>
```
Comment (heathermoses) [inline client/src/components/body/labspage/LabsPage.js]: Could we add just a little bit of padding below the category name?

Screenshot 2024-11-05 at 12 57 15 PM

**Review by heathermoses — CHANGES_REQUESTED** [review]
This looks soooo good, just two small comments!

**Review by heathermoses — APPROVED** [review]
Good stuff Melissa!!

**Review by dm9965 — APPROVED** [review]
Looks great. We'll clean up the wonky lab card later. Awesome job.

**Pull request merged**